### PR TITLE
[CARBONDATA-2593] Add an option 'carbon.insert.storage.level' to support configuring the storage level when insert into data with 'carbon.insert.persist.enable'='true'

### PR DESCRIPTION
**Pull request by zzcclp**
When insert into data with 'carbon.insert.persist.enable'='true', the storage level of dataset is 'MEMORY_AND_DISK',
it should support configuring the storage level to correspond to different environment.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?  No
 
 - [ ] Any backward compatibility impacted?  No
 
 - [ ] Document update required?   No

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6334/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5285/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5172/

**Comment by zzcclp** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6335/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5173/

**Comment by zzcclp** [comment]
@ravipesala @jackylk please help to review, thanks.

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 1]
```diff
@@ -874,6 +874,26 @@
    */
   public static final String CARBON_INSERT_PERSIST_ENABLED_DEFAULT = "false";
 
+  /**
+   * Which storage level to persist dataset when insert into data
+   * with 'carbon.insert.persist.enable'='true'
+   */
+  @CarbonProperty
+  public static final String CARBON_INSERT_STORAGE_LEVEL =
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: Please put an annotation: @InterfaceStability

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 2]
```diff
@@ -874,6 +874,26 @@
...
+      "carbon.insert.storage.level";
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: please add it to document also: https://github.com/apache/carbondata/blob/master/docs/configuration-parameters.md

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java (hunk 1 above)
Comment (zzcclp) [reply core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: Done

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java (hunk 2 above)
Comment (zzcclp) [reply core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: Done

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6351/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5189/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6354/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5192/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5302/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**